pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -103,11 +107,32 @@ func (d *InferenceServiceDefaulter) Default(ctx context.Context, obj runtime.Obj
 	if err != nil {
 		return err
 	}
-	isvc.DefaultInferenceService(configMap, deployConfig)
+
+	var c client.Client
+	if c, err = client.New(cfg, client.Options{Scheme: scheme.Scheme}); err != nil {
+		mutatorLogger.Error(err, "Failed to start client")
+		return err
+	}
+	localModelFound, err := utils.IsCrdAvailable(cfg, v1alpha1.SchemeGroupVersion.String(), constants.ClusterLocalModelKind)
+	if err != nil {
+		mutatorLogger.Error(err, "error when checking if ClusterLocalModel kind is available")
+		return err
+	}
+
+	var models *v1alpha1.ClusterLocalModelList
+	if localModelFound {
```

Comment:
I think we also need to check the localModel enablement flag?

Reply:
hmmm, do you mean a flag on the localModel resource? I haven't added it yet.

Reply:
oh, sorry this means that CR is installed. Renamed the variable.